converter/internal/prometheusconvert/prometheusconvert_test.go

```diff
@@ -2,55 +2,73 @@ package prometheusconvert_test
 
 import (
 	"bytes"
-	"io/fs"
 	"os"
-	"path/filepath"
-	"strings"
 	"testing"
 
 	"github.com/grafana/agent/converter/internal/prometheusconvert"
+	"github.com/grafana/agent/pkg/river/diag"
 	"github.com/stretchr/testify/require"
 )
 
-const (
-	promSuffix = ".yaml"
-	flowSuffix = ".river"
-)
-
 func TestConvert(t *testing.T) {
-	filepath.WalkDir("testdata", func(path string, d fs.DirEntry, _ error) error {
-		if d.IsDir() {
-			return nil
-		}
-
-		if strings.HasSuffix(path, promSuffix) {
-			inputFile := path
-			expectFile := strings.TrimSuffix(path, promSuffix) + flowSuffix
+	tt := []struct {
```

Comment:
Aw man, I kind of liked that test cases were automatically defined before just by adding new files. 

What do you think about having a separate Test for errors, and then the existing gold master tests by using the before/after files?

You could also do something similar to the parser tests, where it looks for a regex in the source file holding an error message to check against, which would help with creating a bunch of these easily without having to have a giant list of defined test cases.

Reply:
To do this via directory/file alone l propose having a third optional TEST.errors file in the testdata directory that has error messages (or not).

Reply:
I have to switch out the river diags for a new error type so it might look a little different in the end, but I have added support for .errors files and put the file driven testing back.